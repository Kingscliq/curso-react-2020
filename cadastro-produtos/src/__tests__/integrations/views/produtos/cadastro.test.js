import "@testing-library/jest-dom";
import { render, act } from "../../../../../test-utils";
import "@testing-library/jest-dom/extend-expect";
import userEvent from "@testing-library/user-event";
import { MemoryRouter } from "react-router-dom";

import Cadastro from "../../../../views/produtos/cadastro";
import React from "react";

let documentBody;

describe("Render correctly <Cadastro />", () => {
  beforeEach(() => {
    documentBody = render(
      <MemoryRouter>
        <Cadastro />
      </MemoryRouter>
    );
  });

  it("should show Nome label", () => {
    const { getByText } = documentBody;
    const label = getByText("Nome: *");
    expect(label).toBeInTheDocument();
  });

  it("should show SKU label", () => {
    const { getByText } = documentBody;
    const label = getByText("SKU: *");
    expect(label).toBeInTheDocument();
  });

  it("should show Descrição label", () => {
    const { getByText } = documentBody;
    const label = getByText("Descrição:");
    expect(label).toBeInTheDocument();
  });

  it("should show Fabricante label", () => {
    const { getByText } = documentBody;
    const label = getByText("Fabricante: *");
    expect(label).toBeInTheDocument();
  });

  it("should show Unidade label", () => {
    const { getByText } = documentBody;
    const label = getByText("Unidade:");
    expect(label).toBeInTheDocument();
  });

  it("should show Preço label", () => {
    const { getByText } = documentBody;
    const label = getByText("Preço: *");
    expect(label).toBeInTheDocument();
  });

  it("should show Fornecedor label", () => {
    const { getByText } = documentBody;
    const label = getByText("Fornecedor: *");
    expect(label).toBeInTheDocument();
  });

  it("should show Custo label", () => {
    const { getByText } = documentBody;
    const label = getByText("Custo:");
    expect(label).toBeInTheDocument();
  });

  it("should click in Salvar button", () => {
    const { getByRole } = documentBody;
    const btn = getByRole("button", {
      name: /Salvar/i,
    });
    act(() => {
      userEvent.click(btn);
    });
  });

  it("should click in Limpar button", () => {
    const { getByRole } = documentBody;
    const btn = getByRole("button", {
      name: /Limpar/i,
    });
    act(() => {
      userEvent.click(btn);
    });
  });

  it("should input text in Nome field", () => {
    const { getByTestId } = documentBody;
    const input = getByTestId("inputNome");
    expect(userEvent.type(input, "Transportadora x")).toBeTruthy();
  });

  it("should input text in SKU field", () => {
    const { getByTestId } = documentBody;
    const input = getByTestId("inputSku");
    expect(userEvent.type(input, "9999")).toBeTruthy();
  });

  it("should input text in Descrição field", () => {
    const { getByTestId } = documentBody;
    const input = getByTestId("inputDescricao");
    expect(userEvent.type(input, "Materiais de construção")).toBeTruthy();
  });

  it("should input text in Fabricante field", () => {
    const { getByTestId } = documentBody;
    const input = getByTestId("inputFabricante");
    expect(userEvent.type(input, "TDL Ltda")).toBeTruthy();
  });

  it("should input text in Unidade field", () => {
    const { getByTestId } = documentBody;
    const input = getByTestId("inputUnidade");
    expect(userEvent.type(input, "25")).toBeTruthy();
  });

  it("should input text in Custo field", () => {
    const { getByTestId } = documentBody;
    const input = getByTestId("inputCusto");
    expect(userEvent.type(input, "32")).toBeTruthy();
  });

  it("should input text in Preco field", () => {
    const { getByTestId } = documentBody;
    const input = getByTestId("inputPreco");
    expect(userEvent.type(input, "20")).toBeTruthy();
  });

  it("should input text in Fornecedor field", () => {
    const { getByTestId } = documentBody;
    const input = getByTestId("inputFornecedor");
    expect(userEvent.type(input, "TAR Company")).toBeTruthy();
  });

  it("should be generate snapshot", () => {
    const { container } = documentBody;
    expect(container).toMatchSnapshot();
  });
});
