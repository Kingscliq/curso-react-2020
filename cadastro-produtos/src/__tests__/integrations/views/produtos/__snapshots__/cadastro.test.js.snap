// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render correctly <Cadastro /> should be generate snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <div
      class="card-header"
    >
      Cadastro de Produto
    </div>
    <div
      class="card-body"
    >
      <form
        id="formProduto"
      >
        <div
          class="row"
        >
          <div
            class="col-md-6"
          >
            <div
              class="form-group"
            >
              <label>
                Nome: *
              </label>
              <input
                class="form-control"
                data-testid="inputNome"
                name="nome"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="col-md-6"
          >
            <div
              class="form-group"
            >
              <label>
                SKU: *
              </label>
              <input
                class="form-control"
                data-testid="inputSku"
                name="sku"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="col-md-12"
          >
            <div
              class="form-group"
            >
              <label>
                Descrição:
              </label>
              <textarea
                class="form-control"
                data-testid="inputDescricao"
                name="descricao"
              />
            </div>
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="col-md-12"
          >
            <div
              class="form-group"
            >
              <label>
                Fabricante: *
              </label>
              <input
                class="form-control"
                data-testid="inputFabricante"
                name="fabricante"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="col-md-6"
          >
            <div
              class="form-group"
            >
              <label>
                Unidade: 
              </label>
              <input
                class="form-control"
                data-testid="inputUnidade"
                name="unidade"
                type="number"
                value="0"
              />
            </div>
          </div>
          <div
            class="col-md-6"
          >
            <div
              class="form-group"
            >
              <label>
                Custo: 
              </label>
              <input
                class="form-control"
                data-testid="inputCusto"
                name="custo"
                type="text"
                value="0"
              />
            </div>
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="col-md-6"
          >
            <div
              class="form-group"
            >
              <label>
                Preço: *
              </label>
              <input
                class="form-control"
                data-testid="inputPreco"
                name="preco"
                type="number"
                value="0"
              />
            </div>
          </div>
          <div
            class="col-md-6"
          >
            <div
              class="form-group"
            >
              <label>
                Fornecedor: *
              </label>
              <input
                class="form-control"
                data-testid="inputFornecedor"
                name="fornecedor"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="col-md-1"
          >
            <button
              class="btn btn-success"
              type="submit"
            >
              Salvar
            </button>
          </div>
          <div
            class="col-md-1"
          >
            <button
              class="btn btn-primary"
            >
              Limpar
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;
